api-put-object.go

```diff
@@ -72,10 +72,10 @@ type PutObjectOptions struct {
 	LegalHold               LegalHoldStatus
 	SendContentMd5          bool
 	DisableMultipart        bool
-	ReplicationVersionID    string
-	ReplicationETag         string
+	SourceVersionID         string
+	SourceETag              string
 	ReplicationStatus       ReplicationStatus
-	ReplicationMTime        time.Time
```

Comment:
Can we simply move this into a separate struct and reference it here. Such as AdvancedOptions - so that you are free to change this struct instead of modifying `PutObjectOptions` on a continuous basis.

It is better to just hide it, add a comment to indicate that not to be used by client applications etc.

Reply:
that's a great idea - done!